addons/base_import/models/base_import.py

```diff
@@ -309,12 +317,12 @@ def do(self, cr, uid, id, fields, options, dryrun=False, context=None):
                   ``false`` if that data isn't available or provided)
         :rtype: list({type, message, record})
         """
-        cr.execute('SAVEPOINT import')
+        self.env.cr.execute('SAVEPOINT import')
 
-        (record,) = self.browse(cr, uid, [id], context=context)
+        (record,) = self
```

Comment:
No need to unpack the record, should use `self.ensure_one()` to make sure there's a single record in the recordset.